test/test_check_branch.py

```diff
@@ -70,3 +70,17 @@ def expect_exception(branch, deployment, message):
                              'prod',
                              "Branch 'feature/foo' cannot be deployed to 'prod', "
                              "only one of {'sandbox'} or personal deployments.")
+
+        variables = ['CI_COMMIT_REF_NAME', 'GITHUB_BASE_REF', 'GITHUB_HEAD_REF']
+
+        for variable in variables:
+            if variable == 'GITHUB_HEAD_REF':
+                branches = ['foo/bar']
+            else:
+                branches = ['develop', 'prod']
+            for branch in branches:
+                with patch.dict(os.environ, {}):
```

Comment:
```suggestion
                with patch.dict(os.environ):
```

Reply:
```suggestion
                with patch.dict(os.environ, {}) as env:
```

and use `env` instead of `os.environ` in the body